### ISSUE TITLE
Cannot use custom attributes with AmplifySignUp React component

### ISSUE THREAD

**Issue by joshgodsiff** [aws-amplify/amplify-js#6510]
**Describe the bug**
I'm trying to add custom attributes to the signup form generated by `AmplifySignUp` from `@aws-amplify/ui-react`.

Additional _standard_ attributes (e.g. `name`) work fine. However, custom attributes (e.g. `custom:business-name`, in my case) do not.

If I attempt to use them in the form, then on form submission I always get an "A client attempted to write unauthorized attribute" error.

From what I can tell from the code, the ability to do this at all may not have yet been ported over from `aws-amplify-react`, but I'm not at all familiar with either codebase, so I may be wrong.

Fields 

**To Reproduce**

1. Set up a bog-standard User Pool and App Client in Cognito
2. Add a custom attribute (`business-name`, in this example).
3. Ensure attribute permissions are correctly set up in 'General Seetings -> App Clients -> Show Details -> Set Attribute Read and Write Permissions', (see screenshot below)
4. Set up a plain create-react-app project with `aws-amplify` and `@aws-amplify/ui-react`
5. Configure Amplify as per below configuration somewhere in your app.
6. Put the below code-snipped in your `App.js` file.

**Expected behavior**
Users are able to sign up.

**Code Snippet**
```
import React from 'react';
import { AmplifySignOut, AmplifyAuthenticator, AmplifySignUp } from '@aws-amplify/ui-react';

const formFields =
  [
    {
      type: "email",
      required: true,
    },
    {
      type: "password",
      required: true,
    },
    {
      type: "phone_number",
      label: 'Phone Number',
      required: false,
      dialCode: '+61'
    },
    {
      type: "name",
      label: "Contact Name",
      placeholder: "e.g. John Smith",
      required: false,
    },
    {
      type: "string",
      label: "Business Name",
      key: 'custom:business-name',
      required: false,
      placeholder: "e.g. ACME Pty Ltd"
    }
  ];

function App() {
  return (
    <div className="App">
      <AmplifyAuthenticator usernameAlias="email">
        <AmplifySignUp formFields={formFields} usernameAlias="email" slot="sign-up" />
        <AmplifySignOut />
        <div>Hello, World</div>
      </AmplifyAuthenticator>
    </div>
  );
}

export default App;
```

**Screenshots**
<img width="1114" alt="Screen Shot 2020-08-06 at 5 00 32 pm" src="https://user-images.githubusercontent.com/145132/89501433-af429080-d806-11ea-8e84-fe01377c01e8.png">

**What is Configured?**
If applicable, please provide what is configured for Amplify CLI:
* Which steps did you follow via Amplify CLI when configuring your resources.
- I am not using the Amplify CLI for configuration. That thing wants far too much in the way of permissions.
* Which resources do you have configured?
   ```
   * If applicable, please provide your manual configuration example:
   ```
{
  "Auth": {
    "region": "ap-southeast-2",
    "userPoolId": "ap-southeast-2_*********",
    "userPoolWebClientId": "**************************",
    "mandatorySignIn": false,
    "oauth": {
      "domain": "****.auth.ap-southeast-2.amazoncognito.com",
      "scope": [
        "phone",
        "email",
        "profile",
        "openid",
        "aws.cognito.signin.user.admin"
      ],
      "redirectSignIn": "http://localhost:3000/",
      "redirectSignOut": "http://localhost:3000/",
      "responseType": "code"
    }
  }
}
   ```
<details>
  <summary><strong>Environment</strong></summary>

<!-- Please run the following command inside your project and copy/paste the output into the codeblock: -->

```
  System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 373.18 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.1.0 - /usr/local/bin/node
    npm: 6.14.5 - /usr/local/bin/npm
  Browsers:
    Chrome: 84.0.4147.105
    Firefox: 78.0.2
    Safari: 13.1.1
  npmPackages:
    @aws-amplify/ui-react: ^0.2.13 => 0.2.13
    @material-ui/core: ^4.11.0 => 4.11.0
    @testing-library/jest-dom: ^4.2.4 => 4.2.4
    @testing-library/react: ^9.5.0 => 9.5.0
    @testing-library/user-event: ^7.2.1 => 7.2.1
    aws-amplify: ^3.0.22 => 3.0.22
    react: ^16.13.1 => 16.13.1
    react-dom: ^16.13.1 => 16.13.1
    react-router-dom: ^5.2.0 => 5.2.0
    react-scripts: 3.4.1 => 3.4.1
  npmGlobalPackages:
    @aws-amplify/cli: 4.26.0
    npm: 6.14.5
```

</details>

**Additional context**
For what it's worth, once the user is logged in, I can edit these fields as you would expect by calling `Auth.updateUserAttributes`, so I don't _think_ it's an issue with my config.

**Comment by jordanranz** [comment]
Hey @joshgodsiff, could you try:
```
{
    type: "custom:business-name",
    label: "Business Name",
    required: false,
    placeholder: "e.g. ACME Pty Ltd"
}
```

**Comment by joshgodsiff** [comment]
That works. 👍 

That could be made a *lot* more explicit / obvious in the documentation. There's a lot of conflicting information around about how to do this - I assume because different libraries have taken different approaches.

**Comment by jordanranz** [comment]
Yep, let me transfer this to the docs repo to track adding custom attribute documentation. Thanks for the feedback!
